### DISCUSSION TITLE
What does 'Changes to this fieldset will stay in sync' mean?

### DISCUSSION THREAD

**Discussion by architchandra** [statamic/cms#6620] [Q&A] [answered]
I see this line when I try to import a field into a blueprint. My understanding based on this is that if I now edit an imported field from inside the blueprint, the changes will also show up in the field under _Fieldsets_. However, #2183 makes it seem that that is not the case. And indeed, if I try it, it's not the case.

What does this mean then? It's not clear to me (for more context, I'm new to Statatmic).
Thank you!

![Screenshot 2022-09-01 at 5 40 03 PM](https://user-images.githubusercontent.com/8830075/187916181-bb78ab82-7cdb-4290-9518-2443598842d6.png)

**Comment by jasonvarga** [comment, marked as answer]
> My understanding based on this is that if I now edit an imported field from inside the blueprint

It's the other way around.

If you were to edit the field from the _fieldset_ the changes would appear wherever you linked it. e.g. in the blueprint.

The fieldset is the source of truth. Then if you edit it from the blueprint, whatever edits you make will only happen at that one spot in the blueprint. It doesn't go back to the fieldset.

**Reply by architchandra** [reply]
Thanks @jasonvarga for the response. I'm still a bit confused, though, because of what I'm seeing at my end.

I have created a _Common_ fieldset that contains a _Bard_ field (called _Body_) and _Grid_ fields called _Image_, _Video_, _Embed_, etc. To maximise re-use, I am importing/linking to these Grid fields from inside the _Body_ field. I am then using this _Body_ field inside a blueprint called _Article_.

![Screenshot 2022-09-04 at 5 19 51 PM](https://user-images.githubusercontent.com/8830075/188312674-5d8ff4f9-8413-432d-977c-8b2cbf99b1af.png)

![Screenshot_2022-09-04_at_5_18_50_PM](https://user-images.githubusercontent.com/8830075/188312727-0d69c8a2-7326-4d08-a5b3-b3075a3afd46.png)

I decided to add a _Caption_ field (text) to the _Video_ Grid field. My expectation was that this change would reflect both in the _Body_ field and inside the _Article_ blueprint automatically, which has not happened.

![Screenshot_2022-09-04_at_5_20_10_PM](https://user-images.githubusercontent.com/8830075/188312772-a552fece-38bb-4046-a9ae-3131cce6a8ae.png)

![Screenshot_2022-09-04_at_5_17_41_PM](https://user-images.githubusercontent.com/8830075/188312830-06ff00a5-480c-4309-af62-7ca6ab7b5ac7.png)

I've tried various combinations of clicking on _Finish_, _Save_ and reloaded the page (more on this in the next comment), but nothing has changed. Is there some command that I need to run to sync the fields?

Thank you so much for taking out the time to help!

**Reply by architchandra** [reply]
An aside is that if I create a field (let's say _Video_), hit 'Save' (_without reloading_ the Fieldset page in the CP) and then go back to my _Body_ (Bard) field, I can't find the newly created _Video_ field in the _Link a single field_ dropdown. It shows up only _after I reload_ the Fieldset page.

Am I missing something?

**Reply by architchandra** [reply]
Update: When I deleted and re-imported the Video field in the Body field, the Article blueprint got updated automatically. So one portion of the puzzle seems to be working. Not sure about the other one (Body field not updating automatically on updating the Video field).

**Reply by architchandra** [reply]
Final update: Because I was adding fields to the _Body_ (Bard) field by linking to other fields, I think I was accidentally making changes in the _Body_ field rather than the original fields in the _Common_ fieldset. Some confusion in this probably resulted in the problems I was facing. Things are working as expected when I make changed directly to the fields inside the Common fieldset. Marking this as answered.
